packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -135,157 +195,199 @@ describe("Presence", () => {
 				});
 
 				describe("that is joining", () => {
-					let newAttendee: ISessionClient | undefined;
-
-					beforeEach(() => {
-						newAttendee = undefined;
-						afterCleanUp.push(
-							presence.events.on("attendeeJoined", (attendee) => {
-								assert(newAttendee === undefined, "Only one attendee should be announced");
-								newAttendee = attendee;
-							}),
-						);
-					});
-
-					function verifyNewAttendee(expectedConnectionId: ClientConnectionId): void {
-						assert(newAttendee !== undefined, "No attendee was announced");
-						assert.equal(
-							newAttendee.sessionId,
-							attendeeSessionId,
-							"Attendee has wrong session id",
-						);
-						assert.equal(
-							newAttendee.getConnectionId(),
-							expectedConnectionId,
-							"Attendee has wrong client connection id",
-						);
-						assert.equal(
-							newAttendee.getConnectionStatus(),
-							SessionClientStatus.Connected,
-							"Attendee connection status is not Connected",
-						);
-					}
-
 					it('first time is announced via `attendeeJoined` with status "Connected"', () => {
 						// Act - simulate join message from client
-						presence.processSignal("", initialAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([initialAttendeeSignalInfo]);
 
 						// Verify
-						verifyNewAttendee(initialAttendeeConnectionId);
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
```

Comment:
We are using `assert.` style elsewhere for verifications. Same here for consistency?

Reply:
Replaced with assert.ok, but not sure that's enough to resolve this. Could also break into multiple assert.strictEquals

Reply:
Asserts should be broken into individual piece checks. When you get a failure, you want to be able to see that the length was the failure or the other check was the failure.